dask/dataframe/core.py

```diff
@@ -6806,9 +6806,15 @@ def to_datetime(arg, meta=None, **kwargs):
             meta = pd.DatetimeIndex([])
             meta.name = arg.name
         else:
-            meta = pd.Series([pd.Timestamp("2000")])
-            meta.index = meta.index.astype(arg.index.dtype)
-            meta.index.name = arg.index.name
+            try:
+                meta = pd.Series([pd.Timestamp("2000")])
+                meta.index = meta.index.astype(arg.index.dtype)
+                meta.index.name = arg.index.name
+            except AttributeError:
```

Comment:
I think we could fail this sooner by catching this case above the `else`.

```python
elif not hasattr(arg, "index"):
    raise ...
```

Reply:
I tried something similar to this at first, but strings still have `hasattr(arg, "index")` evaluate to `True`, that's why I ended up going with catching `AttributeError`. Should we check `not isinstance(arg, [DataFrame, Series])` and raise the `NotImplementedError`?

Reply:
Or `is_dataframe_like` (instead of `isinstance`)

Reply:
I would prefer that personally unless there is some overhead associated.

Reply:
Sounds good. Now raising based on `not (is_dataframe_like(arg) or is_series_like(arg))`